src/modules/stats/reducer.ts

```diff
@@ -0,0 +1,60 @@
+import { loadingReducer, LoadingState } from 'decentraland-dapps/dist/modules/loading/reducer'
+import {
+  FetchSceneStatsFailureAction,
+  FetchSceneStatsRequestAction,
+  FetchSceneStatsSuccessAction,
+  FETCH_SCENE_STATS_FAILURE,
+  FETCH_SCENE_STATS_REQUEST,
+  FETCH_SCENE_STATS_SUCCESS
+} from './action'
+import { WeeklyStats } from './types'
+
+const INITIAL_STATE: StatsState = {
+  data: {},
+  loading: [],
+  error: null
+}
+
+export type StatsState = {
+  data: Record<string, WeeklyStats | null>
+  loading: LoadingState
+  error: string | null
+}
+
+type StatsReducerAction = FetchSceneStatsRequestAction | FetchSceneStatsFailureAction | FetchSceneStatsSuccessAction
+
+export function statsReducer(state = INITIAL_STATE, action: StatsReducerAction) {
+  switch (action.type) {
+    case FETCH_SCENE_STATS_REQUEST: {
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action)
+      }
+    }
+
+    case FETCH_SCENE_STATS_SUCCESS: {
+      const { base, stats } = action.payload
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        error: null,
+        data: {
+          ...state.data,
+          [base]: stats
```

Comment:
This is a new object, but shouldn't we copy it `{ ...stats }` anyways?

Reply:
mm i don't think we do that anywhere tbh, i just checked a few reducers. but i don't see why would you clone it, if we are not mutating?

Reply:
I just checked and I think we are doing it in some cases like asset packs (LOAD_ASSET_PACKS_SUCCESS or LOAD_COLLECTIBLES_SUCCESS) right?

My thought process was that you don't know where else that object is going to go via de action and if you don't clone it here, another part of the store can save it and share a reference

Reply:
i checked and yes, there are a few reducers where we do this, but tbh i don't think we get anything out of breaking a reference if the data hasn't changed. We want to avoid at all costs mutating because that's what breaks pure components and selectors. But if the data is not changing, I don't see why not keep that reference.

For instance in assetPacks we add an `assets` prop to each assetPacks, thus we need to clone it. On collections we omit the `items` prop. But on most of the modules we just assign the instances to the reducer `data`.